[TEST-FAILURE] ClientReplicatedMapListenerTest.testEntryUpdated

```
java.lang.AssertionError: expected:<1> but was:<2>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.client.replicatedmap.ClientReplicatedMapListenerTest$2.run(ClientReplicatedMapListenerTest.java:79)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:827)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:841)
	at com.hazelcast.client.replicatedmap.ClientReplicatedMapListenerTest.testEntryUpdated(ClientReplicatedMapListenerTest.java:76)
```

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x-OpenJDK7/com.hazelcast$hazelcast-client/696/testReport/junit/com.hazelcast.client.replicatedmap/ClientReplicatedMapListenerTest/testEntryUpdated/

--- a/hazelcast-client-legacy/src/main/java/com/hazelcast/client/spi/impl/listener/ClientEventRegistration.java
+++ b/hazelcast-client-legacy/src/main/java/com/hazelcast/client/spi/impl/listener/ClientEventRegistration.java
@@ -18,6 +18,8 @@ package com.hazelcast.client.spi.impl.listener;
 
 import com.hazelcast.nio.Address;
 
+import static com.hazelcast.util.Preconditions.isNotNull;
+
 /**
  * Keeps the information related to to an event registration made by clients
  */
@@ -29,6 +31,7 @@ public class ClientEventRegistration {
 
     public ClientEventRegistration(String serverRegistrationId,
                                    int callId, Address subscriber) {
+        isNotNull(serverRegistrationId, "serverRegistrationId");
         this.serverRegistrationId = serverRegistrationId;
         this.callId = callId;
         this.subscriber = subscriber;
@@ -65,6 +68,26 @@ public class ClientEventRegistration {
      */
     public int getCallId() {
         return callId;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        ClientEventRegistration that = (ClientEventRegistration) o;
+
+        return serverRegistrationId.equals(that.serverRegistrationId);
+
+    }
+
+    @Override
+    public int hashCode() {
+        return serverRegistrationId.hashCode();
     }
 
 }

--- a/hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/listener/ClientEventRegistration.java
+++ b/hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/listener/ClientEventRegistration.java
@@ -19,6 +19,8 @@ package com.hazelcast.client.spi.impl.listener;
 import com.hazelcast.client.spi.impl.ListenerMessageCodec;
 import com.hazelcast.nio.Address;
 
+import static com.hazelcast.util.Preconditions.isNotNull;
+
 /**
  * Keeps the information related to to an event registration made by clients
  */
@@ -31,6 +33,7 @@ public class ClientEventRegistration {
 
     public ClientEventRegistration(String serverRegistrationId,
                                    int callId, Address subscriber, ListenerMessageCodec codec) {
+        isNotNull(serverRegistrationId, "serverRegistrationId");
         this.serverRegistrationId = serverRegistrationId;
         this.callId = callId;
         this.subscriber = subscriber;
@@ -71,6 +74,26 @@ public class ClientEventRegistration {
 
     public ListenerMessageCodec getCodec() {
         return codec;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        ClientEventRegistration that = (ClientEventRegistration) o;
+
+        return serverRegistrationId.equals(that.serverRegistrationId);
+
+    }
+
+    @Override
+    public int hashCode() {
+        return serverRegistrationId.hashCode();
     }
 }
 

